src/rulesets/jsonapi.ts

```diff
@@ -0,0 +1,342 @@
+      "correctly support pagination",
+      (operation, context, docs, specItem) => {
+        if (isOpenApiPath(context.path)) return;
+
+        const operationName = getOperationName(operation);
+
+        const paginationParameters = [
+          "starting_after",
+          "ending_before",
+          "limit",
+        ];
+        const parameterNames = (
+          (specItem.parameters || []) as OpenAPIV3.ParameterObject[]
+        ).map((parameter) => {
+          return parameter.name;
+        });
+        if (!operation.pathPattern.match(/\{[a-z]*?_?id\}$/)) {
+          if (operation.method === "get") {
+            // Require pagination parameters
+            for (const paginationParameterName of paginationParameters) {
+              expect(
+                parameterNames,
+                `expected ${operationName} to include ${paginationParameterName} parameter`,
+              ).to.include(paginationParameterName);
+            }
+            // Require pagination links
+            const response = specItem.responses["200"];
+            if (!("$ref" in response)) {
+              const schema =
+                response.content?.["application/vnd.api+json"]?.schema || {};
+              if (!("$ref" in schema)) {
+                expect(
+                  schema.properties?.links,
+                  `expected ${operationName} to have pagination links`,
+                ).to.exist;
+              }
+            }
+          }
+        } else {
+          if (operation.method !== "get") {
+            for (const paginationParameterName of paginationParameters) {
+              expect(
+                parameterNames,
+                `expected ${operationName} to not include ${paginationParameterName} parameter`,
+              ).to.not.include(paginationParameterName);
+            }
+          }
+        }
+      },
+    );
+  },
+  compoundDocuments: ({ responses }: SnykApiCheckDsl) => {
+    responses.requirement.must(
+      "not allow compound documents",
+      (response, context, docs, specItem) => {
+        if (isOpenApiPath(context.path)) return;
+        if ([200, 201].includes(response.statusCode)) {
+          expect(
+            specItem.content["application/vnd.api+json"]?.schema?.properties
+              ?.included,
+            `expected ${getResponseName(
+              response,
+              context,
+            )} to not support compound documents`,
+          ).to.not.exist;
+        }
+      },
+    );
+  },
+  schemas: ({ responses }: SnykApiCheckDsl) => {
+    responses.requirement.must(
+      "have valid JSON:API schemas",
+      (response, context, docs, specItem) => {
+        if (isOpenApiPath(context.path)) return;
+
+        // Response data
+        if (
+          ["get", "post"].includes(context.method) &&
+          [200, 201].includes(response.statusCode)
+        ) {
+          const responseSchema =
+            specItem.content?.["application/vnd.api+json"]?.schema;
```

Comment:
If the developer mis-types the content type, uses something else like application/json, is the rule error message meaningful enough to the developer, to understand what needs fixing?

Reply:
It made sense in our testing, but it's hard to say what the average developer will know about the guidelines. Do you have an idea what you'd want it to say? 

Also we should go through and add a docsLink for each rule that we can so developers can click into sweater comb docs

Reply:
> It made sense in our testing, but it's hard to say what the average developer will know about the guidelines. Do you have an idea what you'd want it to say?

As I'm writing this, I realize this might be better expressed as a different rule (all 2xx responses except 204 need an `application/vnd.api+json` response), but seeing it referenced here made me wonder how it'd be handled.

> Also we should go through and add a docsLink for each rule that we can so developers can click into sweater comb docs

Yes please, that would be great!

Reply:
Looks like the 2xx rule I'm thinking of is already here: https://github.com/snyk/sweater-comb/pull/99/files#diff-efdbbec0a6118d1d560fdec0ba5b89c1cf08991e76c3f6d8c5143f4229c6c22dR97.